### DISCUSSION TITLE
✅  Option for static files directory

### DISCUSSION THREAD

**Discussion by Isaac-Leonard** [RobinTail/express-zod-api#249] [Ideas]
In express there's a .static() method to specify where plain html / css / frontend js / * files are located, would it be possible to add this to the config object please?

**Comment by RobinTail** [comment]
Well... This method is currently available when using `attachRouting()` with [your own express app](https://github.com/RobinTail/express-zod-api#usage-with-your-own-express-app).
Besides that, after releasing version 5.0.0, `createServer()` also returns `app`... I need to explore my options in this regard.
Thank you @Isaac-Leonard , could be a nice feature.

**Comment by Isaac-Leonard** [comment]
Returning app actually sounds awesome, especially as it seems to solve another idea I had of allowing express-zod-api to work with next's without breaking its auto-deployment feature.
I'll checkout the docs again to see if this will work with it.

**Comment by Isaac-Leonard** [comment]
It seems like this doesn't actually work.
Express-zod-api seems to be catching all routes and returning 404 before static gets reached.

**Comment by RobinTail** [comment]
Thank you for checking this out, @Isaac-Leonard .
Then I will try to find out an elegant approach for implementing [static](http://expressjs.com/en/starter/static-files.html) in the library.

**Comment by Isaac-Leonard** [comment]
A solution might be perhaps only mounting the express-zod-api handler on the actual routes and let express handle the routing so that it doesn't conflict with custom routes attached after express-zod-api has been initialised but then that might be unstable / bug prone.

**Comment by RobinTail** [comment]
I came up with idea to provide a method `serveStatic()` identical to `express.static()` that can be assigned to a path within `Routing`.
https://github.com/RobinTail/express-zod-api/pull/254
This way it enables me to preserve the single place for describing the routes and providing the flexibility of the static files configuration that the original method has.
@Isaac-Leonard , I hope that would fulfill your needs.
```typescript
import {Routing, serveStatic} from "express-zod-api";

const routing: Routing = {
  // path /public serves static files from /example/assets
  public: serveStatic(path.join(__dirname, "assets"), {
    dotfiles: "deny",
    index: false,
    redirect: false,
  }),
};
```

**Comment by RobinTail** [comment]
Finally I made a decision to change it slightly and make it a class, so it will be also consistent with usage of `DependsOsMethods` within `Routing`. The final example looks the following way

```typescript
import {Routing, ServeStatic} from "express-zod-api";

const routing: Routing = {
  // path /public serves static files from ./assets
  public: new ServeStatic(path.join(__dirname, "assets"), {
    dotfiles: "deny",
    index: false,
    redirect: false,
  }),
};
```

**Reply by RobinTail** [reply]
the feature has been released in version 5.2.0 🚀 
Happy new year, @Isaac-Leonard !
